### ISSUE TITLE
Plan an Arquillian 2.0 that will move to Java SE 21 and drop SecurityManager usage

### ISSUE THREAD

**Issue by starksm64** [arquillian/arquillian-core#547]
##### Issue Overview

We should plan for a new major release to base on Java SE 21 and remove all SecurityManager usage.

##### Additional Information

Beyond the SecurityManager deprecation, let's look at what other features can be used to improve the codebase.

**Comment by nlisker** [comment]
Here is what I think should be added to ARQ 2.0 (https://github.com/orgs/arquillian/projects/5):
- https://github.com/arquillian/arquillian-core/issues/772 and https://github.com/arquillian/arquillian-core/pull/854: Cleanup of the behavior of JUnit lifecycle methods with appropriate use of `@RunAsClient` for client context, and server context as default.
- https://github.com/arquillian/arquillian-core/issues/860
- https://github.com/arquillian/arquillian-core/issues/837
- https://github.com/arquillian/arquillian-core/issues/850 (also in https://github.com/arquillian/arquillian-core/issues/144)

**Comment by rhusar** [comment]
@nlisker I cherish the excitement for 2.0! 

However, as discuss in random places, the focus is now to get JUnit Jupiter support fixed (and a big thank you for helping there!). Unless a miracle happens and and a miraculous contributor steps up to supporting multiple branches. :-)

BTW The https://github.com/arquillian/arquillian-core/issues/837 can be implemented in compatible way in a micro and same for https://github.com/arquillian/arquillian-core/issues/860.

**Comment by nlisker** [comment]
The list if what I think should be included when ARQ 2 comes out, not that it's the first release to have those. As you said, some of these can be added in ARQ 1 and therefore included in ARQ 2 when it branches.

I also didn't mean work on ARQ 2 should start now. It's best to fix what's possible in ARQ 1 and then start with ARQ 2. The effort of concurrent support for both versions will depend on how many architectural changes are made,

#850 Can also be added in a micro I'd imagine. The only breaking behavioral changes in the list are the lifecycle annotations for Jupiter.

**Comment by rhusar** [comment]
Exactly; agreed on all points, as that's what I been trying to communicate in the community. Sounds like great past time over upcoming Xmas vacation!
